boards/DIM/Inc/App/App_SevenSegDisplays.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+struct SevenSegDisplay;
+
+// Do NOT take ownership of individual displays
+struct SevenSegDisplays *App_SevenSegDisplays_Create(
```

Comment:
I was having a hard time deciding what should be put into this struct.

```
DIM-9 | 7-segment | - The DIM must display the SoC as percentage on the 7-segment displays if no faults are present.  - If a fault has occurred the DIM must stop displaying the SoC and instead display any faults onto the 7-segment displays. - The first 7-segment must display the board ID while the remaining two must display the fault ID.   - If there are more than one fault active, the DIM must cycle through displaying each present fault at 1Hz.
```

I thought, the DIM must display the SoC so I will add a getter function for that. However, the 7-seg displays must also be smart enough to display SoC instead of faults when needed. How should I be tweaking `struct SevenSegDisplays` for it to be smart enough?

Reply:
I think eventually I want to remove `WriteStateOfCharge` and just have `Tick` function

Reply:
If this is an abstraction around the 7-segment display I don't think it should know anything about the SoC. In general I would expect to see something like:
```
App_SevenSegDisplay_SetValue(seven_seg_display, App_SoC_GetSoCAsString(soc_struct));
```
probably in the state machine (somewhat pseudo-code, but you get the idea). So really the SevenSeg just can set any characters we give it, and we get the SoC as a string and set it. In the case where there's a fault we should just set the corresponding characters. It probably makes sense to have some sort of abstraction around setting faults, just not in the SevenSeg code. 

More generally, the point I'm trying to make here isn't about interface details, but more about what information the SevenSeg logically encapsulates.

Does that make sense? 


edit: After looking through this more, a more reasonable interface might be `App_SevenSegDisplay(struct SevenSeg, HexDigit chars[], size_t num_chars)`.

Reply:
I think I get what you're saying. Just one small question though: Is `App_SevenSegDisplay(struct SevenSeg, HexDigit chars[], size_t num_chars)` supposed to represent all three 7-segment displays? And we would have to check in the function that num_chars is `<= 3` because we don't have the hardware to write more than three characters?

Reply:
I took my own assumptions and wrote 

```
void App_SevenSegDisplays_SetValue(
    const struct SevenSegDisplays *seven_seg_displays,
    uint8_t                        chars[],
    size_t                         num_chars);
```

Feel free to take a look when you get the chance. I also took this function and tested every one of the possible 16 * 16 * 16 = 4096 values.